docs/wiki/TestingUsage.md

```diff
@@ -36,14 +39,53 @@ Invoke-Pester -Configuration @{
 }
 ```
 
-### Handling Resource IDs or Parameters that require or contain Subscription IDs
+## Tool: Use The Test-ModuleLocally Script To Perform Pester Testing, Token Replacement and Deployment of the Module.
+
+The below example can be used on your local environment to load the 'Test-ModuleLocally.ps1' script and modify the required parameters as below to enable you to perform the tests locally before pushing to source control.
+
+```powershell
+#########[ Function Test-ModulesLocally.ps1 ]#############
+
+. 'C:\Users\user\github\user\ResourceModules\utilities\tools\Test-ModuleLocally.ps1'
+
+#########[ Function Test-ModulesLocally.ps1 ]#############
+
+. 'C:\Users\user\github\user\ResourceModules\utilities\tools\Test-ModuleLocally.ps1'
+
+# REQUIRED INPUT FOR TESTING
+$TestModuleLocallyInput = @{
+    ModuleName                    = 'Microsoft.Network\applicationSecurityGroups'
+    PesterTest                    = $true
+    DeploymentTest                = $true
+    ValidationTest                = $false
+    ValidateOrDeployParameters    = @{
+        Location          = 'australiaeast'
+        ResourceGroupName = 'validation-rg'
+        SubscriptionId    = '12345678-1234-1234-1234-123456789123'
+        ManagementGroupId = 'mg-contoso'
+        RemoveDeployment  = $false
+    }
+    DeployAllModuleParameterFiles = $false
+    GetParameterFileTokens        = $true
+    #TokenKeyVaultName             = 'contoso-platform-kv'
+    OtherCustomParameterFileTokens      = @(
+        @{ Name = 'deploymentSpId'; Value = '12345678-1234-1234-1234-123456789123' }
+        @{ Name = 'tenantId'; Value = '12345678-1234-1234-1234-123456789123' }
+    )
+}
+
+```
+
+### Handling Parameters that require or contain a value that should be tokenized
+
+The following scenarios are common to when to use a token value in the parameter file. Refer to [Pipeline Design](./ParameterFileTokens.md) for more details.
 
 - Scenarios where resources have dependencies on other resources, which may require to be linked using `resourceId` references. [Example](../../arm/Microsoft.Network/virtualNetworksResources/virtualNetworkPeerings/.parameters/parameters.json)
```

Comment:
Relative links only works when its relative within the docs folder when this is published into the wiki. Use public link.

Reply:
Fixed